grammar/rainerscript.c

```diff
@@ -1291,7 +1291,7 @@ str2num(es_str_t *s, int *bSuccess)
 	}
 	if(c[0] == '-') {
 		neg = -1;
-		i = -1;
```

Comment:
This seems to be a bug! 

`i` gets set to `-1`, and then is used in the while loop below - reading outside the bounds of the array.  The result is that it cannot parse negative numbers. 

I'm pretty sure this should be `1`, not `-1`.

Reply:
created https://github.com/rsyslog/rsyslog/issues/2255